File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = dict()
        for index, num in enumerate(nums):
            key = target - num
            if key in seen:
                return [seen[key], index]
            else:
                seen[num] = index
                
        return [-1, -1]
                
        

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {

        HashMap<Integer, Integer> seen = new HashMap<>();
        int[] result = {-1, -1};

        for(int i=0; i<nums.length; i++) {
            int key = target-nums[i];
            if (seen.containsKey(key)) {
                result[0] = seen.get(key);
                result[1] = i;
                return result;
            }
            seen.put(nums[i], i);
        }

        return result;
        
    }
}